src/GraphQL/Extensions/SchemaExtensions.cs

```diff
@@ -405,6 +410,91 @@ private void Replace(IProvideResolvedType provider)
+                default:
+                    throw new InvalidOperationException($"The '{import.Name}' import name is not valid; please specify only '@link', 'Purpose', and 'Import'.");
+            }
+        }
+        schema.LinkSchema(config);
+        var linkPurposeGraphType = new LinkPurposeGraphType() { Name = config.NameForType("Purpose") };
+        var linkImportGraphType = new LinkImportGraphType() { Name = config.NameForType("Import") };
+        var linkDirective = new Directive("link")
+        {
+            Arguments = new QueryArguments(
+                new QueryArgument(typeof(NonNullGraphType<StringGraphType>)) { Name = "url" },
+                new QueryArgument(typeof(StringGraphType)) { Name = "as" },
+                new QueryArgument(linkPurposeGraphType) { Name = "purpose" },
+                new QueryArgument(new ListGraphType(linkImportGraphType)) { Name = "import" }),
+            Repeatable = true,
+        };
+        linkDirective.Locations.Add(DirectiveLocation.Schema);
+        schema.Directives.Register(linkDirective);
+    }
+
+    /// <summary>
+    /// Adds a @link directive to the schema, specifying a URL and optional configuration for the link.
+    /// </summary>
+    /// <param name="schema">The schema to which the directive will be added.</param>
+    /// <param name="url">The URL of the linked schema.</param>
+    /// <param name="configuration">An optional action to configure the link further.</param>
+    private static void LinkSchema(this ISchema schema, string url, Action<LinkConfiguration>? configuration = null)
+    {
+        var config = new LinkConfiguration(url);
+        configuration?.Invoke(config);
+        schema.LinkSchema(config);
+    }
+
+    /// <inheritdoc cref="LinkSchema(ISchema, string, Action{LinkConfiguration}?)"/>
+    private static void LinkSchema(this ISchema schema, LinkConfiguration config)
+    {
+        if (config.Url == null)
+            throw new InvalidOperationException("No URL specified for the linked schema.");
+
+        var linkInstalled = false;
+        var appliedDirectives = schema.GetAppliedDirectives();
+        if (appliedDirectives != null)
+        {
+            foreach (var appliedDirective in appliedDirectives)
+            {
+                if (appliedDirective.Name == "link")
+                {
+                    var url = appliedDirective.FindArgument("url")?.Value as string;
+                    if (url == config.Url)
+                        throw new InvalidOperationException($"The schema is already linked to the schema at '{config.Url}'.");
+                    if (url == LinkConfiguration.LINK_URL)
+                        linkInstalled = true;
+                }
+            }
+        }
+        if (!linkInstalled && config.Url != LinkConfiguration.LINK_URL)
+        {
+            schema.AddLinkDirectiveSupport();
+        }
+
+        schema.ApplyDirective("link", config.ConfigureAppliedDirective);
+    }
```

Comment:
I don't understand the code in this class. `AddLinkDirectiveSupport` is the only new public method, and it's called from the private `LinkSchema`, which is called from the public `AddLinkDirectiveSupport`...
Also I don't see any usage of the other private `LinkSchema` overload (with `url`).
If the `LinkConfiguration.LINK_URL` is the only URL currently supported, why is there an option to configure it in this version?
`LINK_URL` points to `v1.0`, so no support for `v2.0`?
I'm somewhat confused...

Reply:
See below. Only version that exists is 1.0 currently. 

https://specs.apollo.dev/link/v1.0/

Reply:
It's a `@link` spec version, not a federation version 🤦‍♂️

Reply:
Wait. Shouldn't it be a federation spec URL?
```
extend schema
  @link(url: "https://specs.apollo.dev/federation/v2.3",
        import: ["@key", "@shareable"])
```
https://www.apollographql.com/docs/federation/federated-types/federated-directives/

Reply:
The goal is to support linking in the federation spec. First it links the link spec though because that’s not part of the federation spec. So each federated schema is going to have two links on it.